ckanext/harvest/logic/action/update.py

```diff
@@ -462,16 +462,16 @@ def harvest_send_job_to_gather_queue(context, data_dict):
     :type id: string
     '''
     log.info('Send job to gather queue: %r', data_dict)
-    check_access('harvest_send_job_to_gather_queue', context, data_dict)
 
     job_id = logic.get_or_bust(data_dict, 'id')
+    job = toolkit.get_action('harvest_job_show')(
+        context, {'id': job_id})
+
+    check_access('harvest_send_job_to_gather_queue', context, job)
```

Comment:
harvest_send_job_to_gather_queue needs to know the source_id, so bring the harvest_job_show call earlier and pass in the result.